extmod/modtls_mbedtls.c

```diff
@@ -603,6 +708,22 @@ static mp_obj_t ssl_socket_make_new(mp_obj_ssl_context_t *ssl_context, mp_obj_t
 
     mbedtls_ssl_init(&o->ssl);
 
+    // Configure PSK if enabled
+    if (ssl_context->use_psk && ssl_context->psk_identity != mp_const_none && ssl_context->psk_key != mp_const_none) {
+        // Get PSK identity and key
+        size_t psk_identity_len;
+        const byte *psk_identity = (const byte *)mp_obj_str_get_data(ssl_context->psk_identity, &psk_identity_len);
+        
+        size_t psk_key_len;
+        const byte *psk_key = (const byte *)mp_obj_str_get_data(ssl_context->psk_key, &psk_key_len);
```

Comment:
These conversions could raise a `bad_implicit_conversion` that would bypass cleanup --- neither this method nor the setters for those variables actually check `mp_obj_is_str_or_bytes`.